src/coreclr/src/vm/dllimportcallback.cpp

```diff
@@ -961,12 +961,18 @@ void UMEntryThunk::Terminate()
     CONTRACTL
     {
         NOTHROW;
+        MODE_ANY;
     }
     CONTRACTL_END;
 
     m_code.Poison();
 
     s_thunkFreeList.AddToList(this);
+
+    if (GetObjectHandle())
```

Comment:
It maybe also nice to set the object handle field to zero when we destroy the handle.